### DISCUSSION TITLE
Kalman Filter initialization Configuration

### DISCUSSION THREAD

**Discussion by GNC-JY** [bitcraze/discussions#1013] [Q&A] [answered]
Hello.

I found a singularity while using the Kalman Filter of Crazyflie 2.1 and asked questions.

While setting the estimator to kalman and checking the values of roll, pitch, I started the crazyflie on a flat surface when I wasn't flying, and I saw it reset the attitude estimate again where it tilted.

In the code related to the kalman filter, is there a code that initializes the attitude estimate when the accel sensor value or gyro sensor value is close to zero or sensors estimate value is a problem?

**Comment by krichardsson** [comment]
The kalman estimator is automatically reset when it is in a state that is unlikely to be correct, currently I think it is when it is more than 100 m from the origin or the velocity is higher than 10 m/s. 
If no positioning system is used (some form of external position sensor input), the kalman estimator usually drifts away due to the noise in the IMU and will reset at regular intervals. There will be a line printed in the console log when this happens.
I'm not sure if this is what you see?

**Reply by GNC-JY** [reply]
Can you reply where is code?

**Reply by krichardsson** [reply, marked as answer]
The out of bounds check is done in [kalman_supervisor.c](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/modules/src/kalman_supervisor.c)
It is called from [estimator_kalman.c](https://github.com/bitcraze/crazyflie-firmware/blob/dbb9df1137f11d4e7e3771c56d25a7137b5b69cc/src/modules/src/estimator/estimator_kalman.c#L257-L264)
